analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldVisible.cs

```diff
@@ -46,21 +46,22 @@ public sealed class StaticFieldVisible : SonarDiagnosticAnalyzer
                 c =>
                 {
                     var fieldDeclaration = (FieldDeclarationSyntax)c.Node;
-                    foreach (var field in fieldDeclaration.Declaration.Variables
-                        .Select(variableDeclaratorSyntax => new
-                        {
-                            Syntax = variableDeclaratorSyntax,
-                            Symbol = c.SemanticModel.GetDeclaredSymbol(variableDeclaratorSyntax) as IFieldSymbol
-                        })
-                        .Where(f => FieldIsRelevant(f.Symbol)))
+                    foreach (var diagnostic in GetDiagnostics(fieldDeclaration.Declaration, c.SemanticModel))
                     {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, field.Syntax.Identifier.GetLocation(), field.Syntax.Identifier.ValueText));
+                        c.ReportDiagnosticWhenActive(diagnostic);
                     }
                 },
                 SyntaxKind.FieldDeclaration);
 
+        private static IEnumerable<Diagnostic> GetDiagnostics(VariableDeclarationSyntax declaration, SemanticModel semanticModel) =>
+            declaration.Variables
+                       .Select(declarator => new {Syntax = declarator, Symbol = semanticModel.GetDeclaredSymbol(declarator) as IFieldSymbol})
+                       .Where(f => FieldIsRelevant(f.Symbol))
+                       .Select(pair => Diagnostic.Create(Rule, pair.Syntax.Identifier.GetLocation(), pair.Syntax.Identifier.ValueText));
```

Comment:
It's weird to name the same thing `f` on one line and `pair` on another. My personal favorite is `x`, as nothing special is needed to distinguish them